src/main/java/cd/go/contrib/elasticagents/docker/Agent.java

```diff
@@ -103,6 +103,10 @@ public static String toJSONArray(Collection<Agent> metadata) {
         return new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create().toJson(metadata);
     }
 
+    public void setElasticAgentId(String elasticAgentId) {
```

Comment:
Exposing setter is not good idea. For this particular scenario I will go with constructor. Once you initialize agent no one can change the agen id

Reply:
Agreed. Do u recommend any other default values for the other agent parameters? Setting AgentId alone does not define an agent completely. @bdpiparva